endpoints/globus/endpoint.go

```diff
@@ -316,159 +325,161 @@ func (ep *Endpoint) Transfer(dst core.Endpoint, files []core.FileTransfer) (uuid
+		Data:                xferItems,
+		DestinationEndpoint: gDst.Id.String(),
+		SourceEndpoint:      ep.Id.String(),
+		SyncLevel:           3, // transfer only if checksums don't match
+		VerifyChecksum:      true,
+		FailOnQuotaErrors:   true,
+	})
+	if err == nil {
+		return xferId, err
+	}
+	resp, err = ep.post("transfer", bytes.NewReader(data))
+	if err != nil {
+		return xferId, err
+	}
+	defer resp.Body.Close()
+	body, err = io.ReadAll(resp.Body)
+	if err != nil {
+		return xferId, err
+	}
+	type SubmissionResponse struct {
+		TaskId  uuid.UUID `json:"task_id"`
+		Code    string    `json:"code"`
+		Message string    `json:"message"`
+	}
+
+	var gResp SubmissionResponse
+	err = json.Unmarshal(body, &gResp)
+	if err != nil {
+		return xferId, err
+	}
+	xferId = gResp.TaskId
+	var zeroId uuid.UUID
+	if xferId == zeroId { // trouble!
+		return xferId, fmt.Errorf("%s (%s)", gResp.Message, gResp.Code)
+	}
+	return xferId, nil
 }
 
 func (ep *Endpoint) Status(id uuid.UUID) (core.TransferStatus, error) {
 	resource := fmt.Sprintf("task/%s", id.String())
 	resp, err := ep.get(resource, url.Values{})
-	if err == nil {
-		defer resp.Body.Close()
-		var body []byte
-		body, err = io.ReadAll(resp.Body)
-		if err == nil {
-			type TaskResponse struct {
-				Files            int    `json:"files"`
-				FilesTransferred int    `json:"files_transferred"`
-				IsPaused         bool   `json:"is_paused"`
-				Status           string `json:"status"`
-				Code             string `json:"code"`
-				Message          string `json:"message"`
-			}
-			var response TaskResponse
-			err = json.Unmarshal(body, &response)
-			if err == nil {
-				if strings.Contains(response.Code, "ClientError") {
-					return core.TransferStatus{}, fmt.Errorf(response.Message)
-				} else {
-					codes := map[string]core.TransferStatusCode{
-						"Active":    core.TransferStatusActive,
-						"Inactive":  core.TransferStatusInactive,
-						"Succeeded": core.TransferStatusSucceeded,
-						"Failed":    core.TransferStatusFailed,
-					}
-					return core.TransferStatus{
-						Code:                codes[response.Status],
-						NumFiles:            response.Files,
-						NumFilesTransferred: response.FilesTransferred,
-					}, nil
-				}
-			}
+	if err != nil {
+		return core.TransferStatus{}, err
+	}
+	defer resp.Body.Close()
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return core.TransferStatus{}, err
+	}
+	type TaskResponse struct {
+		Files            int    `json:"files"`
+		FilesTransferred int    `json:"files_transferred"`
+		IsPaused         bool   `json:"is_paused"`
+		Status           string `json:"status"`
+		Code             string `json:"code"`
+		Message          string `json:"message"`
+	}
+	var response TaskResponse
+	err = json.Unmarshal(body, &response)
+	if err != nil {
+		return core.TransferStatus{}, err
+	}
+	if strings.Contains(response.Code, "ClientError") {
+		return core.TransferStatus{}, fmt.Errorf(response.Message)
+	} else {
```

Comment:
Not really worth changing as it's readable as-is, but this could be flattened out a bit more as there's no need for the `else` when the `if` `return`s a result.